---
title: "Demultiplex With fastq-multx"
params:
  atacama_data_percent: 1
  data_dir : /data
  atacama_ps_rds: /data/tutorial_data/atacama_1pct.rds
  
output:
  md_document:
    variant: markdown_github
  html_document:
    df_print: paged
---


# Setup

First we load libraries.
```{r libraries, message=FALSE, warning=FALSE}
library(readr)
library(fs)
library(dplyr)
library(tibble)
library(Biostrings)
# library(R.utils)
```

## Paths, Directories, and Shell Variables
To keep the code readable and portable, it is nice to assign paths to variables.  We also need to use the R `Sys.setenv` command to make shell variables that can be used in the bash chunks below.



```{r files_and_directories, include=FALSE}
# Params

if(exists("params") && 
   !is.null(params[["atacama_data_percent"]])){
  atacama.data.percent=params[["atacama_data_percent"]]
} else {
  atacama.data.percent = "1"
}

if(exists("params") && 
   !is.null(params[["data_dir"]])){
  data.parent.dir=params[["data_dir"]]
} else {
  data.parent.dir = "/data"
}

data.dir=file.path(data.parent.dir,"tutorial_data")





# Directories
data.dir = file.path(data.dir, paste0("atacama_", atacama.data.percent, "pct"))
output.dir = path.expand(file.path("~/scratch", paste0("atacama_", atacama.data.percent, "pct")))
demux.dir = file.path(output.dir, "demux")

# make directory for output
if (dir_exists(demux.dir)) {dir_delete(demux.dir)}
dir_create(demux.dir)

# Files
barcode.table = file.path(output.dir,"barcodes_for_fastqmultx.tsv")
rc.barcode.table = file.path(output.dir,"rc_barcodes_for_fastqmultx.tsv")

map.file = file.path(data.dir,"sample_metadata.tsv")
barcode.fastq = file.path(data.dir,"barcodes.fastq.gz")
r1.fastq = file.path(data.dir,"forward.fastq.gz")
r2.fastq = file.path(data.dir,"reverse.fastq.gz")

# Set variables for bash
Sys.setenv(MAP_FILE = map.file)
Sys.setenv(OUT_DIR = output.dir)
Sys.setenv(DEMUX_DIR = demux.dir)
Sys.setenv(RAW_FASTQ_DIR = data.dir)
Sys.setenv(BARCODE_TABLE = barcode.table)
Sys.setenv(RC_BARCODE_TABLE = rc.barcode.table)
Sys.setenv(BARCODE_FASTQ = barcode.fastq)
Sys.setenv(R1_FASTQ = r1.fastq)
Sys.setenv(R2_FASTQ = r2.fastq)
```

## Check Data Integrity
```{bash check_md5sum}
cd $RAW_FASTQ_DIR
md5sum -c md5sum.txt
```


# Demultiplexing
We will be using `fastq-multx` to demultiplex the data.  It does not have very good documentation.  But, we can get some instructions if we run it without any command line options.
```{bash error=TRUE}
fastq-multx
```

## Barcode Table
The `fastq-multx` help tells us that we can supply it with a tab-separated file specifying the sample ID in the first column and the barcode sequence that corresponds with it in the second column.  We can easily generate a file like this from our map file using the unix `cut` command.  The purpose of `cut` is to extract specific columns from files.  The sample IDs and barcode sequences are in columns (fields) 1 and 2 of the map file.  Here is the command that we want to use.
```{bash}
set -u
cut --fields 1,2 $MAP_FILE > $BARCODE_TABLE
```

Now let's take a peek at the barcode tsv that we have generated to be sure it looks as expected.
```{bash}
set -u
head $BARCODE_TABLE
```

## Running fastq-multx
We now have everything we need to run `fastq-multx`.  Here is an explanation for the command line options that we use

-m : number of mismatches to allow in barcode (relative to the barcode supplied in the barcode table)
-x : don’t trim barcodes (this isn't necessary)
-B BARCODE_FILE : a list of known barcodes, and the associated sample names
-d : minimum edit distance between the best and next best match
BARCODE_FASTQ : the index read FASTQ, which will be used to demultiplex other reads
R1_FASTQ : the R1 raw data to demultiplex
R2_FASTQ : (optional) if data is paired-end, the R2 raw data to demultiplex

-o OUTPUT_FILE(s) : fastq-multx will produce a separate file for each barcode (two files when paired-end reads are input, three files when barcodes are in a separate I1 FASTQ). This option provides a template for naming the output file - the program will fill in the “%” with the barcode.

Because of the way `fastq-multx` is designed (to allow demuxing of FASTQs that have the barcode embeded in the R1 sequence), it will actually demux the I1 FASTQ.  Since we are only interesed in the R1 and R2, we can ignore the demuxed I1 files.
```{bash}
set -u
fastq-multx -m 3 -d 2 -x -B $BARCODE_TABLE \
  $BARCODE_FASTQ \
  $R1_FASTQ \
  $R2_FASTQ \
  -o $DEMUX_DIR/%_I1.fastq.gz \
  -o $DEMUX_DIR/%.forward.fastq.gz \
  -o $DEMUX_DIR/%.reverse.fastq.gz
```

## The Unmatched Problem
### Reverse Complement the barcodes
```{r}
read_tsv(map.file, comment= "#q2") %>% # read in map file, dropping the line that starts with "#q2"
  select(Sample = "#SampleID", 
         BarcodeSequence) %>%          # select only the columns with sample ID (renamed to get rid of "#") and the barcode itself
  deframe %>%                          # convert to a named vector (expected input for DNAStringSet constructor)
  DNAStringSet %>%                     # convert to DNAStringSet
  reverseComplement %>%                # reverse complement the barcodes
  as.data.frame %>%                    # convert to dataframe for write_delim
  rownames_to_column %>% 
  write_delim(rc.barcode.table,        # output barcodes to a file 
              delim="\t", 
              col_names=FALSE)

# clean up the output from the previous demultiplexing step
if (dir_exists(demux.dir)) {dir_delete(demux.dir)}
dir_create(demux.dir)
```

### Run Demux with RC barcodes
```{bash}
set -u
fastq-multx -m 3 -d 2 -x -B $RC_BARCODE_TABLE \
  $BARCODE_FASTQ \
  $R1_FASTQ \
  $R2_FASTQ \
  -o $DEMUX_DIR/%_I1.fastq.gz \
  -o $DEMUX_DIR/%.forward.fastq.gz \
  -o $DEMUX_DIR/%.reverse.fastq.gz
```

# Session Info
Always print `sessionInfo` for reproducibility!
```{r}
sessionInfo()
```

